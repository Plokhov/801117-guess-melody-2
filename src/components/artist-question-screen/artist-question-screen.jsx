import React from "react";
import PropTypes from "prop-types";

import {AudioPlayer} from "../audio-player/audio-player.jsx";

export class ArtistQuestionScreen extends React.PureComponent {
  constructor(props) {
    super(props);

    this.state = {
      isPlaying: false,
    };
  }
  render() {
    const {question, onAnswer, screenIndex} = this.props;
    const {isPlaying} = this.state;
    const {
      answers,
      song,
    } = question;

    return <section className="game__screen">
      <h2 className="game__title">
        Кто исполняет эту песню?
      </h2>
      <div className="game__track">
        <AudioPlayer
          isPlaying={isPlaying}
          onPlayButtonClick={() => this.setState({isPlaying: !isPlaying})}
          src={song.src}
        />
      </div>

      <form className="game__artist">
        {answers.map((it) => {
          return (
            <div
              className="artist"
              key={`${screenIndex} - answer-${it.id}`}
            >
              <input
                className="artist__input visually-hidden"
                type="radio"
                name="answer"
                value={`artist-${it.id}`}
                id={`artist-${it.id}`}
                onClick={() => {
                  onAnswer(it);
                  this.setState({
                    isPlaying: false,
                  });
                }}
              />
              <label className="artist__name" htmlFor={`artist-${it.id}`}>
                <img
                  className="artist__picture"
                  src={it.picture}
                  alt={it.artist}
                />
                {it.artist}
              </label>
            </div>);
        })}
      </form>
    </section>;
  }
}

ArtistQuestionScreen.propTypes = {
  question: PropTypes.shape({
    type: PropTypes.oneOf([`genre`, `artist`]).isRequired,
    song: PropTypes.shape({
      artist: PropTypes.string.isRequired,
      src: PropTypes.string.isRequired
    }),
    answers: PropTypes.arrayOf(PropTypes.shape({
      id: PropTypes.string.isRequired,
      picture: PropTypes.string.isRequired,
      artist: PropTypes.string.isRequired
    })),
  }).isRequired,
  screenIndex: PropTypes.number.isRequired,
  onAnswer: PropTypes.func.isRequired,
};
