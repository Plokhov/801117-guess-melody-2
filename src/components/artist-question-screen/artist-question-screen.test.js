import React from 'react';
import renderer from 'react-test-renderer';
import {ArtistQuestionScreen} from './artist-question-screen.jsx';

it(`ArtistQuestionScreen correctly renders after relaunch`, () => {
  const tree = renderer
    .create(<ArtistQuestionScreen
      screenIndex={0}
      question={{
        type: `a`,
        song: {
          artist: `a`,
          src: `a`,
        },
        answers: [
          {
            picture: `a`,
            artist: `a`,
          },
          {
            picture: `b`,
            artist: `b`,
          },
          {
            picture: `c`,
            artist: `c`,
          },
        ],
      }}
      onAnswer={jest.fn()}
    />)
    .toJSON();

  expect(tree).toMatchSnapshot();
});
