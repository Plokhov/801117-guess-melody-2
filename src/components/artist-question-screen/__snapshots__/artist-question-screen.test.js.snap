// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArtistQuestionScreen correctly renders after relaunch 1`] = `
<section
  className="game game--artist"
>
  <header
    className="game__header"
  >
    <a
      className="game__back"
    >
      <span
        className="visually-hidden"
      >
        Сыграть ещё раз
      </span>
      <img
        alt="Угадай мелодию"
        className="game__logo"
        src="img/melody-logo-ginger.png"
      />
    </a>
    <div
      className="timer__value"
    >
      <span
        className="timer__mins"
      >
        05
      </span>
      <span
        className="timer__dots"
      >
        :
      </span>
      <span
        className="timer__secs"
      >
        00
      </span>
    </div>
    <div
      className="game__mistakes"
    >
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
      <div
        className="wrong"
      />
    </div>
  </header>
  <section
    className="game__screen"
  >
    <h2
      className="game__title"
    >
      Кто исполняет эту песню?
    </h2>
    <div
      className="game__track"
    >
      <div
        className="track"
      >
        <button
          className="track__button track__button--play"
          type="button"
        />
        <div
          className="track__status"
        >
          <audio />
        </div>
      </div>
    </div>
    <form
      className="game__artist"
      onChange={[MockFunction]}
    >
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-1"
          name="answer"
          type="radio"
          value="answer-1"
        />
        <label
          className="artist__name"
          htmlFor="answer-1"
        >
          <img
            alt="a"
            className="artist__picture"
            src="a"
          />
          a
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-2"
          name="answer"
          type="radio"
          value="answer-2"
        />
        <label
          className="artist__name"
          htmlFor="answer-2"
        >
          <img
            alt="b"
            className="artist__picture"
            src="b"
          />
          b
        </label>
      </div>
      <div
        className="artist"
      >
        <input
          className="artist__input visually-hidden"
          id="answer-3"
          name="answer"
          type="radio"
          value="answer-3"
        />
        <label
          className="artist__name"
          htmlFor="answer-3"
        >
          <img
            alt="c"
            className="artist__picture"
            src="c"
          />
          c
        </label>
      </div>
    </form>
  </section>
</section>
`;
